diff --git a/Mkfiles/msvc.mak b/Mkfiles/msvc.mak
index b743412..239fe3a 100644
--- a/Mkfiles/msvc.mak
+++ b/Mkfiles/msvc.mak
@@ -116,7 +116,19 @@ version.mac: version version.pl
 macros.c: macros.pl pptok.ph standard.mac version.mac \
 	$(srcdir)/macros/*.mac $(srcdir)/output/*.mac
 	$(PERL) $(srcdir)/macros.pl $(srcdir)/standard.mac version.mac \
-		$(srcdir)/macros/*.mac $(srcdir)/output/*.mac
+		$(srcdir)/macros/altreg.mac \
+		$(srcdir)/macros/fp.mac \
+		$(srcdir)/macros/ifunc.mac \
+		$(srcdir)/macros/smartalign.mac \
+		$(srcdir)/output/outaout.mac \
+		$(srcdir)/output/outas86.mac \
+		$(srcdir)/output/outbin.mac \
+		$(srcdir)/output/outcoff.mac \
+		$(srcdir)/output/outelf.mac \
+		$(srcdir)/output/outmacho.mac \
+		$(srcdir)/output/outobj.mac \
+		$(srcdir)/output/outrdf.mac \
+		$(srcdir)/output/outrdf2.mac
 
 # These source files are generated from regs.dat by yet another
 # perl script.
diff --git a/ndisasm.c b/ndisasm.c
index 754033c..07ffe57 100644
--- a/ndisasm.c
+++ b/ndisasm.c
@@ -346,27 +346,12 @@ static void output_ins(uint32_t offset, uint8_t *data,
                        int datalen, char *insn)
 {
     int bytes;
-    fprintf(stdout, "%08"PRIX32"  ", offset);
+    fprintf(stdout, "/* %08"PRIx32" %s */\n", offset, insn);
 
-    bytes = 0;
-    while (datalen > 0 && bytes < BPL) {
-        fprintf(stdout, "%02X", *data++);
-        bytes++;
-        datalen--;
-    }
-
-    fprintf(stdout, "%*s%s\n", (BPL + 1 - bytes) * 2, "", insn);
-
-    while (datalen > 0) {
-        fprintf(stdout, "         -");
-        bytes = 0;
-        while (datalen > 0 && bytes < BPL) {
-            fprintf(stdout, "%02X", *data++);
-            bytes++;
-            datalen--;
-        }
-        fprintf(stdout, "\n");
+    for (bytes = 0; bytes < datalen; bytes++) {
+        fprintf(stdout, "0x%02x, ", data[bytes]);
     }
+    fprintf(stdout, "\n");
 }
 
 /*
